### PR TITLE
[posix] support interface name in ot-ctl

### PR DESCRIPTION
**Pull request by bukepo**
This commit adds the Thread interface name to unix socket of OpenThread daemon, so that `ot-ctl` can specify which daemon to connect to.

### CODE REVIEW

**Comment by bukepo** [comment]
will update after #6557 is merged.

File: src/posix/platform/system.cpp [hunk 1]
```diff
@@ -162,6 +162,8 @@ otInstance *otSysInit(otPlatformConfig *aPlatformConfig)
     platformNetifInit(instance, aPlatformConfig->mInterfaceName);
 #elif OPENTHREAD_CONFIG_PLATFORM_UDP_ENABLE
     platformUdpInit(aPlatformConfig->mInterfaceName);
+#else
+    gNetifName[0] = '\0';
```
Comment (simonlingoogle) [inline src/posix/platform/system.cpp:166]: Why not initialize gNetifName with "" ?

File: src/posix/client.cpp [hunk 2]
```diff
@@ -161,28 +175,82 @@ static bool ReconnectSession(void)
...
+        }
+    }
+
+    aArgCount -= optind;
+    aArgVector += optind;
+
+    return config;
+}
+
+extern "C" int main(int argc, char *argv[])
```
Comment (simonlingoogle) [inline src/posix/client.cpp:234]: why do we need `extern "C"` ?

File: src/posix/client.cpp [hunk 3]
```diff
@@ -111,7 +120,7 @@ static bool DoWrite(int aFile, const void *aBuffer, size_t aSize)
     return ret;
 }
 
-static int ConnectSession(void)
```
Comment (simonlingoogle) [inline src/posix/client.cpp:114]: why do we remove `static`?

File: src/posix/platform/system.cpp (hunk 1 above)
Comment (bukepo) [reply src/posix/platform/system.cpp:166]: Ensure it's initialized.

File: src/posix/client.cpp (hunk 3 above)
Comment (bukepo) [reply src/posix/client.cpp:114]: We use `static` to avoid global function conflicts. Now they are all inside of `ot::Client` namespace.

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1: 
with questions

File: src/posix/client.cpp (hunk 2 above)
Comment (bukepo) [reply src/posix/client.cpp:234]: save `ot::Client::` for function calls.

**Review by jwhui — APPROVED** [review]

**Pull request merged**